python/ray/train/tests/test_huggingface_gpu.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import tempfile
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pytest
+from ray.train.batch_predictor import BatchPredictor
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+
+
+import ray
+from ray.train.huggingface import HuggingFaceCheckpoint, HuggingFacePredictor
+
+test_strings = ["Complete me", "And me", "Please complete"]
+prompts = pd.DataFrame(test_strings, columns=["sentences"])
+
+# We are only testing Causal Language Modeling here
+
+model_checkpoint = "hf-internal-testing/tiny-random-gpt2"
+tokenizer_checkpoint = "hf-internal-testing/tiny-random-gpt2"
+
+
+def create_checkpoint():
+    with tempfile.TemporaryDirectory() as tmpdir:
+        model_config = AutoConfig.from_pretrained(model_checkpoint)
+        model = AutoModelForCausalLM.from_config(model_config)
+        tokenizer = AutoTokenizer.from_pretrained(tokenizer_checkpoint)
+        checkpoint = HuggingFaceCheckpoint.from_model(model, tokenizer, path=tmpdir)
+        # Serialize to dict so we can remove the temporary directory
+        return HuggingFaceCheckpoint.from_dict(checkpoint.to_dict())
+
+
+@pytest.mark.parametrize("batch_type", [pd.DataFrame])
+@pytest.mark.parametrize("device", [None, 0])
+def test_predict_batch(ray_start_4_cpus, caplog, batch_type, device):
+    checkpoint = create_checkpoint()
+    predictor = BatchPredictor.from_checkpoint(
+        checkpoint, HuggingFacePredictor, task="text-generation"
+    )
+
+    # Todo: Ray data does not support numpy string arrays well
+    if batch_type == np.ndarray:
+        dataset = ray.data.from_numpy(prompts.to_numpy().astype("U"))
+    elif batch_type == pd.DataFrame:
+        dataset = ray.data.from_pandas(prompts)
+    elif batch_type == pa.Table:
```

Comment:
let's remove arrow if it's no longer being supported?

Reply:
It's in multiple tests - we should do this in a separate PR